torchci/scripts/update_test_times.py

```diff
@@ -0,0 +1,55 @@
+from collections import defaultdict
+import json
+import rockset
+import os
+from pathlib import Path
+
+REPO_ROOT = Path(__file__).resolve().parent.parent.parent
+
+PROD_VERSIONS_FILE = REPO_ROOT / "torchci" / "rockset" / "prodVersions.json"
+
+
+def main() -> None:
+    rockset_client = rockset.RocksetClient(
+        host="api.usw2a1.rockset.com", api_key=os.environ["ROCKSET_API_KEY"]
+    )
+    with open(PROD_VERSIONS_FILE) as f:
+        prod_versions = json.load(f)
+
+    rockset_result = rockset_client.QueryLambdas.execute_query_lambda(
+        query_lambda="test_time_per_file",
+        version=prod_versions["commons"]["test_time_per_file"],
+        workspace="commons",
+    ).results
+    periodic_rockset_result = rockset_client.QueryLambdas.execute_query_lambda(
+        query_lambda="test_time_per_file_periodic_jobs",
+        version=prod_versions["commons"]["test_time_per_file_periodic_jobs"],
+        workspace="commons",
+    ).results
+
+    test_times = defaultdict(lambda: defaultdict(lambda: defaultdict(float)))
+    test_times_no_build_env = defaultdict(lambda: defaultdict(list))
+    test_times_no_test_config = defaultdict(list)
+    for row in rockset_result + periodic_rockset_result:
+        test_times[row["base_name"]][row["test_config"]][row["file"]] = row["time"]
+        test_times_no_build_env[row["test_config"]][row["file"]].append(row["time"])
+        test_times_no_test_config[row["file"]].append(row["time"])
+
+    # Add defaults
+    for config in test_times_no_build_env:
+        for test, times in test_times_no_build_env[config].items():
+            test_times_no_build_env[config][test] = sum(times) / len(times)
```

Comment:
Does `test_times_no_build_env` overwrite itself here,? i.e. `test_times_no_build_env[config][test]` was a list of test time, then it becomes the average `sum(times) / len(times)`.  Same goes for `test_times_no_test_config`.  It's probably not a big deal, just makes me feel a bit uneasy.

May be a good idea would be to add some unit tests for the part from line 30 to line 48.

Reply:
Yeah it edits in place, I'm also not a big fan, but it means I don't have to come up with another variable name

Added tests and changed the workflow file to run those tests